### DISCUSSION TITLE
omnet++ pro exe file

### DISCUSSION THREAD

**Discussion by Aib-star** [omnetpp/omnetpp#992] [Q&A] [answered]
I want to build the project and run it by running the generated exe file. My project relies on inet. However, after running the exe file, tkenv shows that the ned in inet cannot be found.

**Comment by aarizaq** [comment]
I am not sure if this will work, but it is possible to include a symbolic link in your directory to the dll file.

**Reply by Aib-star** [reply]
thank u very much! which directory? Can you explain that in detail?thank u again

**Comment by aarizaq** [comment]
First of all, the tkenv has been removed since omnet++ 5 and replaced by qtenv, I suppose that you are using omnet 4. If you launch the simulation using omnet++ IDE (if the installation is correct) The IDE should find the libraries.
If you launch the simulation using the command line it is possible that the exe file could not fund the dll libraries. In this case, it is possible to copy the dll files or include symbolic links in the directory to the dll file.

**Reply by Aib-star** [reply]
Sorry, I didn't explain the environment.

I am using omnetpp 5.6.1, inet4.2 and win11.
I want to use the command line to move the project. The ide running project is successful.

I have tried to copy and paste the libinet.dll file of inet's library file into my project directory, it doesn't work.
My question is how to include symbolic links in the directory to the dll file.

**Reply by Aib-star** [reply]
I used the command line to make a Symbolic link: mklink/D "libINET" "D:  omnetpp-5.6.1  samples  inet4  out  clang release  src  libINET. dll"
But the project cannot run normally, with an error: Loading NED files from.:
<!> Error: Could not load NED sources from '.': Cannot temporarily change to directory 'libINET' (does it exist?)
I confirm that libINET.dll exists in this path
How should this be resolved?

**Comment by Aib-star** [comment]
Sorry, I didn't explain the environment.

I am using omnetpp 5.6.1, inet4.2 and win11.  
I want to use the command line to move the project. The ide running project is successful.

I have tried to copy and paste the libinet.dll file of inet's library file into my project directory, it doesn't work. 
My question is how to include symbolic links in the directory to the dll file.
---- Replied Message ----
| From | ***@***.***> |
| Date | 06/10/2023 01:17 |
| To | ***@***.***> |
| Cc | Aib-star ***@***.***>,
Author ***@***.***> |
| Subject | Re: [omnetpp/omnetpp] omnet++ pro exe file (Discussion #992) |

First of all, the tkenv has been removed since omnet++ 5 and replaced by qtenv, I suppose that you are using omnet 4. If you launch the simulation using omnet++ IDE (if the installation is correct) The IDE should find the libraries.
If you launch the simulation using the command line it is possible that the exe file could not fund the dll libraries. In this case, it is possible to copy the dll files or include symbolic links in the directory to the dll file.

**Comment by aarizaq** [comment]
when you start the from the omnet command line you can set the libraries path

For example, this is the command lone for an example in my computer.
$ cd /home/alfonso/modelos/openflow/scenarios/hyperflow/fattree
$ ./fattree.exe -m -n ../../../src:../..:../../../../inetmanet-4.x/examples:../../../../inetmanet-4.x/showcases:../../../../inetmanet-4.x/src:../../../../inetmanet-4.x/tests/networks:../../../../inetmanet-4.x/tests/validation:../../../../inetmanet-4.x/tutorials -x inet.showcases.visualizer.osg;inet.transportlayer.tcp_lwip;inet.visualizer.osg --image-path=../../../images:../../../../inetmanet-4.x/images -l ../../../src/openflow6 -l ../../../../inetmanet-4.x/src/INET Scenario_DynamicFatTree_HF_Ping_Flood_4C.ini

If you launch the simulation in the IDE, you can see the command that you need to use shell in the console tab
-l ../../../../inetmanet-4.x/src/INET  is the patht to the inet dll library


________________________________
De: Aib-star ***@***.***>
Enviado: sábado, 10 de junio de 2023 1:32
Para: omnetpp/omnetpp ***@***.***>
Cc: Alfonso ***@***.***>; Comment ***@***.***>
Asunto: Re: [omnetpp/omnetpp] omnet++ pro exe file (Discussion #992)


Sorry, I didn't explain the environment.

I am using omnetpp 5.6.1, inet4.2 and win11.
I want to use the command line to move the project. The ide running project is successful.

I have tried to copy and paste the libinet.dll file of inet's library file into my project directory, it doesn't work.
My question is how to include symbolic links in the directory to the dll file.
---- Replied Message ----
| From | ***@***.***> |
| Date | 06/10/2023 01:17 |
| To | ***@***.***> |
| Cc | Aib-star ***@***.***>,
Author ***@***.***> |
| Subject | Re: [omnetpp/omnetpp] omnet++ pro exe file (Discussion #992) |

First of all, the tkenv has been removed since omnet++ 5 and replaced by qtenv, I suppose that you are using omnet 4. If you launch the simulation using omnet++ IDE (if the installation is correct) The IDE should find the libraries.
If you launch the simulation using the command line it is possible that the exe file could not fund the dll libraries. In this case, it is possible to copy the dll files or include symbolic links in the directory to the dll file.

**Reply by Aib-star** [reply, marked as answer]
Thank u very much!I have already resolved this issue this afternoon. I specified the path to ned in the PATH(winows) and the path to the inet library file in the ini file, it works. Thanks again!
